### PR TITLE
Resize join block automatically

### PR DESCRIPTION
**Pull request by balderdash**
Depends on https://github.com/code-dot-org/blockly/pull/14

### CODE REVIEW

File: apps/src/blocksCommon.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ function installJoinBlock(blockly) {
     },
 
     setInputCount: function (inputCount) {
-      var newInputCount = parseInt(inputCount);
+      var newInputCount = Math.max(parseInt(inputCount), 2);
```
Comment (Hamms) [inline apps/src/blocksCommon.js:193]: it's not clear to me why this is not dangerous; is there no situation in which we want to set inputcount to 1?

File: apps/src/blocksCommon.js [hunk 2]
```diff
@@ -205,6 +205,39 @@ function installJoinBlock(blockly) {
       }
       this.inputCount = newInputCount;
     },
+
+    pendingConnection: function (oldConnection, newConnection) {
+      var lastConnectionIndex = 0;
+      var oldConnectionIndex = -1;
+      var newConnectionIndex = -1;
+      for (var i = 0; i < this.inputCount; i++) {
```
Comment (Hamms) [inline apps/src/blocksCommon.js]: why `inputCount` rather than `this.inputList.length`?

File: apps/src/blocksCommon.js [hunk 3]
```diff
@@ -205,6 +205,39 @@ function installJoinBlock(blockly) {
...
+        }
+        if (connection === newConnection) {
+          newConnectionIndex = i;
+        }
+      }
+
+      var toEnd = newConnectionIndex >= lastConnectionIndex;
+      var fromEnd = oldConnectionIndex >= lastConnectionIndex;
+
+      if (this.delayedResize && toEnd ^ fromEnd) {
```
Comment (Hamms) [inline apps/src/blocksCommon.js:229]: an XOR encountered in the wild! 😮 It's beautiful.

File: apps/src/blocksCommon.js [hunk 4]
```diff
@@ -205,6 +205,39 @@ function installJoinBlock(blockly) {
...
+        window.clearTimeout(this.delayedResize);
+        this.delayedResize = null;
+      }
+      if (toEnd && !fromEnd) {
+        this.setInputCount(lastConnectionIndex + 2);
+      } else if (fromEnd && !toEnd) {
+        this.delayedResize = window.setTimeout(function () {
+          this.setInputCount(lastConnectionIndex + 1);
+        }.bind(this), 100);
```
Comment (Hamms) [inline apps/src/blocksCommon.js]: nit: could use an arrow function

File: apps/src/blocksCommon.js (hunk 1 above)
Comment (Hamms) [reply apps/src/blocksCommon.js:193]: after manual review, I can see that there is not.

File: apps/src/blocksCommon.js (hunk 2 above)
Comment (balderdash) [reply apps/src/blocksCommon.js]: They *should* always be the same, but `this.inputList.length` makes more sense.

File: apps/src/blocksCommon.js (hunk 4 above)
Comment (balderdash) [reply apps/src/blocksCommon.js]: Oh yeah, done.

**Comment by balderdash** [comment]
As discussed offline, since the existing block isn't yet used anywhere, I'm going to stick with modifying it instead of making a new block.

**Pull request merged**